File: examples/python/edit/transform_map.py
import os
import numpy as np
import pywavemap as wave

# Load the map
user_home = os.path.expanduser('~')
input_map_path = os.path.join(user_home, "your_map.wvmp")
your_map = wave.Map.load(input_map_path)

# Define a transformation that flips the map upside down, for illustration
translation = np.array([0.0, 0.0, 0.0])
rotation = wave.Rotation(w=0.0, x=1.0, y=0.0, z=0.0)
transformation = wave.Pose(rotation, translation)

# Transform the map
your_map = wave.edit.transform(your_map, transformation)

# Save the map
output_map_path = os.path.join(user_home, "your_map_transformed.wvmp")
your_map.store(output_map_path)

File: examples/cpp/edit/transform_map.cc
#include <memory>

#include <wavemap/core/common.h>
#include <wavemap/core/utils/edit/transform.h>
#include <wavemap/io/file_conversions.h>

using namespace wavemap;  // NOLINT
int main(int, char**) {
  // Load the map
  const std::filesystem::path home_dir = CHECK_NOTNULL(getenv("HOME"));
  const std::filesystem::path input_map_path = home_dir / "your_map.wvmp";
  MapBase::Ptr map_base;
  bool success = io::fileToMap(input_map_path, map_base);
  CHECK(success);

  // Downcast it to a concrete map type
  auto map = std::dynamic_pointer_cast<HashedWaveletOctree>(map_base);
  CHECK_NOTNULL(map);

  // Define a transformation that flips the map upside down, for illustration
  Transformation3D T_AB;
  T_AB.getRotation() = Rotation3D{0.f, 1.f, 0.f, 0.f};

  // Transform the map
  auto thread_pool = std::make_shared<ThreadPool>();  // Optional
  map = edit::transform(*map, T_AB, thread_pool);

  // Save the map
  const std::filesystem::path output_map_path =
      home_dir / "your_map_transformed.wvmp";
  success &= io::mapToFile(*map, output_map_path);
  CHECK(success);
}
